### PR TITLE
lookatme: init at 2.3.2

### PR DESCRIPTION
**Pull request by ameertaweel**
###### Motivation for this change

This is a very useful package.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by legendofmiracles — CHANGES_REQUESTED** [review]
This is not the right way to package a python app, I've [written a patch](http://0x0.st/oXZV.txt) to make the code more idiomatic

File: pkgs/top-level/all-packages.nix
Comment (legendofmiracles) [inline pkgs/top-level/all-packages.nix:34755]: I didn't spot this before, all-packages is sorted alphabetically and not which package is the newest

File: maintainers/maintainer-list.nix
Comment (AmeerTaweel) [reply maintainers/maintainer-list.nix:13832]: done

File: pkgs/top-level/all-packages.nix
Comment (AmeerTaweel) [reply pkgs/top-level/all-packages.nix:34755]: done

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 158323` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>lookatme</li>
  </ul>
</details>

File: pkgs/tools/misc/lookatme/default.nix
Comment (06kellyjac) [inline pkgs/tools/misc/lookatme/default.nix:8]: @AmeerTaweel first, welcome. :rocket:  Secondly, one thing to note is we're getting rid of python 2 (https://github.com/NixOS/nixpkgs/issues/148779) so i'd either:

create a click package that's v7 using python 3 (this might help: https://github.com/NixOS/nixpkgs/blob/6bdc4f3fe2ba3ed340b84b3b715013e07937c892/pkgs/development/python-modules/click/default.nix#L1)

or you'd need to update lookatme upstream to use click v8. Click v8 came out 2021-05-11 so it's probably about time :sweat_smile:

Comment (AmeerTaweel) [reply pkgs/tools/misc/lookatme/default.nix:8]: I want to submit a pull-request to make upstream use `click v8`. This is my idea on how to do it:

+ Update `requirements.txt` of the package.
+ Update the Nix derivation.
+ Build and test.
+ Submit pull-request to upstream.
+ They (hopefully) merge it.
+ I submit the updated Nix derivation.

Is that good? Or am I missing some thing(s)?

Comment (AmeerTaweel) [reply pkgs/tools/misc/lookatme/default.nix:8]: Of course, I'm assuming that doing the update will not break anything. If it breaks something, I will have to fix it.

Comment (06kellyjac) [reply pkgs/tools/misc/lookatme/default.nix:8]: Yeah that sounds good. I'm not sure how `buildPythonApplication`/`fetchPypi` work exactly but you can probably change `src =` to be `src = ./your-repo-with-changes;`

I'd note a major version change from v7 to v8 probably has some breaking changes that'll need some modifications in the python code, especially since the requirements.txt explicitly says not v8 currently.

For submitting the updated nix derivation you could either pull the existing 2.3.2 version and patch it, or wait for 2.3.3 with your changes. (example of a patch fixing something ahead of a new release https://github.com/NixOS/nixpkgs/pull/158707/files#diff-e8c4f9d22def9371b17eb15d3f5df0dd00e177ff96deea05a265222880ec7cb6R15-R22)

Comment (AmeerTaweel) [reply pkgs/tools/misc/lookatme/default.nix:8]: Thanks for your help. Will open an issue upstream to inform the authors about what I want to do.

Comment (AmeerTaweel) [reply pkgs/tools/misc/lookatme/default.nix:8]: Made an [issue](https://github.com/d0c-s4vage/lookatme/issues/131) upstream and waiting for approval.

Comment (AmeerTaweel) [reply pkgs/tools/misc/lookatme/default.nix:8]: @06kellyjac it's been about two months since I created the issue but got no response at all. What should I do?
Should I forget about that and try the first option of `click v7` with `python3`?

Comment (06kellyjac) [reply pkgs/tools/misc/lookatme/default.nix:8]: The lookatme project doesn't seem to have made any movement in a long time.

The preferred solution would be to update lookatme. We can then package it with the patch file from the PR until that's merged in.

You could try maintain a copy of click v7 just for use by lookatme in nixpkgs but at some point we do cull old versions of software.

I think you'll need to review if lookatme is important enough for you to put in the effort to make this PR bumping to click v8.
It might even come down to you forking the project and maintaining it yourself since the main dev doesn't look active in any projects anymore

Comment (AmeerTaweel) [reply pkgs/tools/misc/lookatme/default.nix:8]: Actually it's not that important to me. I use it for my presentations, but to be honest, it does not worth the effort...

Comment (06kellyjac) [reply pkgs/tools/misc/lookatme/default.nix:8]: Yeah, makes sense.
Pip install or a virtual env might work. Or you can run it in a container

**Pull request merged**
